--- origsrc/gobject-introspection-0.10.6/Makefile-gir.am
+++ src/gobject-introspection-0.10.6/Makefile-gir.am
@@ -15,17 +15,17 @@ STATIC_GIRSOURCES =               \
 EXTRA_DIST += $(STATIC_GIRSOURCES)
 
 # Doesn't seem to work to do gir/%.typelib =(
-gir/cairo-1.0.typelib: g-ir-compiler
-gir/DBus-1.0.typelib: g-ir-compiler
-gir/DBusGLib-1.0.typelib: g-ir-compiler
-gir/fontconfig-2.0.typelib: g-ir-compiler
-gir/freetype2-2.0.typelib: g-ir-compiler
-gir/GL-1.0.typelib: g-ir-compiler
-gir/libxml2-2.0.typelib: g-ir-compiler
-gir/xft-2.0.typelib: g-ir-compiler
-gir/xlib-2.0.typelib: g-ir-compiler
-gir/xfixes-4.0.typelib: g-ir-compiler
-gir/xrandr-1.3.typelib: g-ir-compiler
+gir/cairo-1.0.typelib: g-ir-compiler$(EXEEXT)
+gir/DBus-1.0.typelib: g-ir-compiler$(EXEEXT)
+gir/DBusGLib-1.0.typelib: g-ir-compiler$(EXEEXT)
+gir/fontconfig-2.0.typelib: g-ir-compiler$(EXEEXT)
+gir/freetype2-2.0.typelib: g-ir-compiler$(EXEEXT)
+gir/GL-1.0.typelib: g-ir-compiler$(EXEEXT)
+gir/libxml2-2.0.typelib: g-ir-compiler$(EXEEXT)
+gir/xft-2.0.typelib: g-ir-compiler$(EXEEXT)
+gir/xlib-2.0.typelib: g-ir-compiler$(EXEEXT)
+gir/xfixes-4.0.typelib: g-ir-compiler$(EXEEXT)
+gir/xrandr-1.3.typelib: g-ir-compiler$(EXEEXT)
 
 # glib with source (to enable documentation)
 if WITH_GLIBSRC
@@ -85,7 +85,7 @@ GLib_2_0_gir_FILES =  \
 
 BUILT_GIRSOURCES += GLib-2.0.gir
 
-GLib-2.0.gir: g-ir-scanner g-ir-compiler
+GLib-2.0.gir: g-ir-scanner g-ir-compiler$(EXEEXT)
 
 gir/DBusGLib-1.0.typelib: GObject-2.0.gir
 
--- origsrc/gobject-introspection-0.10.3/common.mk	2010-11-11 08:58:33.000000000 -0600
+++ src/gobject-introspection-0.10.3/common.mk	2011-03-02 20:22:13.446982100 -0600
@@ -11,7 +11,7 @@ INTROSPECTION_SCANNER = \
         PYTHONPATH=$(top_builddir):$(top_srcdir) \
 	UNINSTALLED_INTROSPECTION_SRCDIR=$(top_srcdir) \
 	UNINSTALLED_INTROSPECTION_BUILDDIR=$(top_builddir) \
-    $(top_builddir)/g-ir-scanner$(EXEEXT)
+    $(top_builddir)/g-ir-scanner
 INTROSPECTION_SCANNER_ARGS = \
     --verbose \
     -I$(top_srcdir) \
--- origsrc/gobject-introspection-0.10.3/gobject-introspection-1.0.pc.in	2010-10-29 12:36:40.000000000 -0500
+++ src/gobject-introspection-0.10.3/gobject-introspection-1.0.pc.in	2011-03-02 20:22:50.512646800 -0600
@@ -7,8 +7,8 @@ datadir=@datadir@
 includedir=@includedir@
 
 g_ir_scanner=${bindir}/g-ir-scanner
-g_ir_compiler=${bindir}/g-ir-compiler
-g_ir_generate=${bindir}/g-ir-generate
+g_ir_compiler=${bindir}/g-ir-compiler@EXEEXT@
+g_ir_generate=${bindir}/g-ir-generate@EXEEXT@
 gidatadir=${datadir}/gobject-introspection-1.0
 girdir=${datadir}/gir-1.0
 typelibdir=${libdir}/girepository-1.0
--- origsrc/gobject-introspection-0.10.3/gobject-introspection-no-export-1.0.pc.in	2010-09-02 08:41:11.000000000 -0500
+++ src/gobject-introspection-0.10.3/gobject-introspection-no-export-1.0.pc.in	2011-03-02 20:22:55.941456200 -0600
@@ -7,8 +7,8 @@ datadir=@datadir@
 includedir=@includedir@
 
 g_ir_scanner=${bindir}/g-ir-scanner
-g_ir_compiler=${bindir}/g-ir-compiler
-g_ir_generate=${bindir}/g-ir-generate
+g_ir_compiler=${bindir}/g-ir-compiler@EXEEXT@
+g_ir_generate=${bindir}/g-ir-generate@EXEEXT@
 girdir=${datadir}/gir-1.0
 typelibdir=${libdir}/girepository-1.0
 
