--- origsrc/gobject-introspection-0.10.7/docs/reference/Makefile.am	2010-11-17 15:36:22.000000000 -0600
+++ src/gobject-introspection-0.10.7/docs/reference/Makefile.am	2011-04-06 05:12:55.384373300 -0500
@@ -7,7 +7,7 @@ DOC_MODULE=gi
 DOC_MAIN_SGML_FILE=gi-docs.sgml
 
 # The directory containing the source code. Relative to $(srcdir)
-DOC_SOURCE_DIR=../../girepository
+DOC_SOURCE_DIR=$(top_srcdir)/girepository
 
 # Extra options to supply to gtkdoc-scan
 SCAN_OPTIONS=
