--- origsrc/gobject-introspection-0.10.3/Makefile-giscanner.am	2010-11-09 12:57:01.000000000 -0600
+++ src/gobject-introspection-0.10.3/Makefile-giscanner.am	2011-03-02 20:25:05.811682700 -0600
@@ -64,6 +64,8 @@ if OS_WIN32
 # Yuck. Probably there is a way to get this from Python, but I don't
 # know how. Use -Wl to avoid libtool crack.
 _giscanner_la_LDFLAGS += -Wl,$(pyexecdir)/../../libs/libpython25.a -no-undefined
+else
+_giscanner_la_LDFLAGS += -L$(libdir)/python$(PYTHON_VERSION)/config -lpython$(PYTHON_VERSION) -no-undefined
 endif
 
 _giscanner_la_SOURCES = giscanner/giscannermodule.c
--- origsrc/gobject-introspection-0.10.3/tests/Makefile.am	2011-01-20 08:33:07.000000000 -0600
+++ src/gobject-introspection-0.10.3/tests/Makefile.am	2011-03-02 20:25:05.827282700 -0600
@@ -7,7 +7,6 @@ BUILT_SOURCES=
 CLEANFILES=
 
 INCLUDES = $(GOBJECT_CFLAGS)
-LIBADD = $(GOBJECT_LIBS)
 
 testsdir = $(datadir)/gobject-introspection-1.0/tests
 tests_DATA =		    \
@@ -21,7 +20,12 @@ testlibdir=$(prefix)/unused
 install-testlibLTLIBRARIES: # prevent it from being installed
 
 libeverything_1_0_la_SOURCES = everything.c
+libeverything_1_0_la_LDFLAGS = -no-undefined
+libeverything_1_0_la_LIBADD = $(GOBJECT_LIBS)
+
 libgimarshallingtests_1_0_la_SOURCES = gimarshallingtests.c
+libgimarshallingtests_1_0_la_LDFLAGS = -no-undefined
+libgimarshallingtests_1_0_la_LIBADD = $(GOBJECT_LIBS)
 
 EXTRA_DIST += gimarshallingtests.h
 
--- origsrc/gobject-introspection-0.10.3/tests/offsets/Makefile.am	2010-11-16 09:07:28.000000000 -0600
+++ src/gobject-introspection-0.10.3/tests/offsets/Makefile.am	2011-03-02 20:25:05.842882700 -0600
@@ -18,7 +18,7 @@ liboffsets_la_SOURCES = \
 	offsets.c
 liboffsets_la_CPPFLAGS = $(GIREPO_CFLAGS)
 # dummy rpath to get built dynamically (huh?)
-liboffsets_la_LDFLAGS = -avoid-version -rpath $(libdir)
+liboffsets_la_LDFLAGS = -avoid-version -rpath $(libdir) -no-undefined
 
 Offsets-1.0.gir: liboffsets.la offsets.h
 Offsets_1_0_gir_INCLUDES = GObject-2.0
--- origsrc/gobject-introspection-0.10.3/tests/scanner/Makefile.am	2011-01-20 08:33:07.000000000 -0600
+++ src/gobject-introspection-0.10.3/tests/scanner/Makefile.am	2011-03-02 20:25:05.842882700 -0600
@@ -20,7 +20,7 @@ install-testlibLTLIBRARIES: # prevent it
 
 AM_CPPFLAGS = -I$(top_srcdir)/girepository
 AM_CFLAGS = $(GIO_CFLAGS) $(GOBJECT_CFLAGS) $(GTHREAD_CFLAGS)
-AM_LDFLAGS = -avoid-version
+AM_LDFLAGS = -avoid-version -no-undefined
 LIBS = $(GOBJECT_LIBS) $(GTHREAD_LIBS)
 
 libannotation_la_SOURCES = $(srcdir)/annotation.c $(srcdir)/annotation.h
@@ -35,13 +35,10 @@ testsdir = $(datadir)/gobject-introspect
 tests_DATA = $(libregress_la_SOURCES)
 
 libregress_la_CFLAGS = $(AM_CFLAGS) $(CAIRO_CFLAGS)
-libregress_la_LDFLAGS = $(AM_LDFLAGS) $(CAIRO_LIBS) 
+libregress_la_LDFLAGS = $(AM_LDFLAGS)
+libregress_la_LIBADD = $(CAIRO_LIBS) $(GIO_LIBS)
 
 
-if OS_WIN32
-AM_LDFLAGS += -no-undefined
-endif
-
 # .gir --[scanner]-> .typelib
 GIRS =
 TYPELIBS = $(GIRS:.gir=.typelib)
@@ -108,9 +105,9 @@ barapp_SOURCES = $(srcdir)/barapp.c $(sr
 barapp_LDADD = $(top_builddir)/libgirepository-1.0.la
 barapp_LDFLAGS = -export-dynamic
 
-Bar-1.0.gir: barapp
+Bar-1.0.gir: barapp$(EXEEXT)
 Bar_1_0_gir_PACKAGES = gobject-2.0
-Bar_1_0_gir_PROGRAM = ./barapp
+Bar_1_0_gir_PROGRAM = ./barapp$(EXEEXT)
 Bar_1_0_gir_INCLUDES = GObject-2.0
 Bar_1_0_gir_FILES = $(barapp_SOURCES)
 Bar_1_0_gir_SCANNERFLAGS = --accept-unprefixed
