--- origsrc/gobject-introspection-0.6.10/gir/Everything-1.0-expected.gir	2010-04-18 13:11:38.000000000 -0500
+++ src/gobject-introspection-0.6.10/gir/Everything-1.0-expected.gir	2010-05-13 00:24:17.683828400 -0500
@@ -12,7 +12,7 @@ and/or use gtk-doc annotations.  -->
   <package name="gobject-2.0"/>
   <namespace name="Everything"
              version="1.0"
-             shared-library="libgirepository-everything-1.0.so.1"
+             shared-library="cyggirepository-everything-1.0-1.dll"
              c:prefix="Everything">
     <record name="TestBoxed"
             c:type="TestBoxed"
--- origsrc/gobject-introspection-0.6.10/tests/scanner/annotation-1.0-expected.gir	2010-01-21 12:36:18.000000000 -0600
+++ src/gobject-introspection-0.6.10/tests/scanner/annotation-1.0-expected.gir	2010-05-13 00:29:38.647186400 -0500
@@ -12,7 +12,7 @@ and/or use gtk-doc annotations.  -->
   <package name="gobject-2.0"/>
   <namespace name="annotation"
              version="1.0"
-             shared-library="libannotation.so"
+             shared-library="cygannotation.dll"
              c:prefix="annotation">
     <callback name="Callback"
               c:type="AnnotationCallback"
--- origsrc/gobject-introspection-0.6.10/tests/scanner/annotation-1.0-expected.tgir	2010-04-18 13:12:57.000000000 -0500
+++ src/gobject-introspection-0.6.10/tests/scanner/annotation-1.0-expected.tgir	2010-05-13 00:30:45.088986700 -0500
@@ -6,7 +6,7 @@
   <include name="utility" version="1.0"/>
   <include name="GObject" version="2.0"/>
   <include name="GLib" version="2.0"/>
-  <namespace name="annotation" version="1.0" shared-library="libannotation.so" c:prefix="annotation">
+  <namespace name="annotation" version="1.0" shared-library="cygannotation.dll" c:prefix="annotation">
     <callback name="Callback">
       <return-value transfer-ownership="none">
         <type name="int"/>
--- origsrc/gobject-introspection-0.6.10/tests/scanner/drawable-1.0-expected.gir	2009-12-16 10:52:43.000000000 -0600
+++ src/gobject-introspection-0.6.10/tests/scanner/drawable-1.0-expected.gir	2010-05-13 00:29:52.967005500 -0500
@@ -12,7 +12,7 @@ and/or use gtk-doc annotations.  -->
   <package name="gobject-2.0"/>
   <namespace name="drawable"
              version="1.0"
-             shared-library="libdrawable.so"
+             shared-library="cygdrawable.dll"
              c:prefix="drawable">
     <class name="TestDrawable"
            c:type="TestDrawable"
--- origsrc/gobject-introspection-0.6.10/tests/scanner/drawable-1.0-expected.tgir	2010-04-18 13:12:58.000000000 -0500
+++ src/gobject-introspection-0.6.10/tests/scanner/drawable-1.0-expected.tgir	2010-05-13 00:30:53.897490500 -0500
@@ -6,7 +6,7 @@
   <include name="utility" version="1.0"/>
   <include name="GObject" version="2.0"/>
   <include name="GLib" version="2.0"/>
-  <namespace name="drawable" version="1.0" shared-library="libdrawable.so" c:prefix="drawable">
+  <namespace name="drawable" version="1.0" shared-library="cygdrawable.dll" c:prefix="drawable">
     <class name="TestDrawable" parent="GObject.Object" glib:type-struct="TestDrawableClass" abstract="1" glib:type-name="TestDrawable" glib:get-type="test_drawable_get_type">
       <field name="parent_instance">
         <type name="GObject.Object"/>
--- origsrc/gobject-introspection-0.6.10/tests/scanner/foo-1.0-expected.gir	2010-03-18 16:43:48.000000000 -0500
+++ src/gobject-introspection-0.6.10/tests/scanner/foo-1.0-expected.gir	2010-05-13 00:30:00.111414100 -0500
@@ -14,7 +14,7 @@ and/or use gtk-doc annotations.  -->
   <c:include name="foo.h"/>
   <namespace name="foo"
              version="1.0"
-             shared-library="libfoo.so"
+             shared-library="cygfoo.dll"
              c:prefix="foo">
     <alias name="List" target="GLib.SList" c:type="FooList"/>
     <alias name="ObjectCookie" target="any" c:type="FooObjectCookie"/>
--- origsrc/gobject-introspection-0.6.10/tests/scanner/foo-1.0-expected.tgir	2010-04-18 13:13:00.000000000 -0500
+++ src/gobject-introspection-0.6.10/tests/scanner/foo-1.0-expected.tgir	2010-05-13 00:31:00.073843700 -0500
@@ -7,7 +7,7 @@
   <include name="Gio" version="2.0"/>
   <include name="GObject" version="2.0"/>
   <include name="GLib" version="2.0"/>
-  <namespace name="foo" version="1.0" shared-library="libfoo.so" c:prefix="foo">
+  <namespace name="foo" version="1.0" shared-library="cygfoo.dll" c:prefix="foo">
     <enumeration name="ASingle">
       <member name="some_single_enum" value="0"/>
     </enumeration>
--- origsrc/gobject-introspection-0.6.10/tests/scanner/GtkFrob-1.0-expected.gir	2009-09-17 03:53:45.000000000 -0500
+++ src/gobject-introspection-0.6.10/tests/scanner/GtkFrob-1.0-expected.gir	2010-05-13 00:30:06.839798900 -0500
@@ -11,7 +11,7 @@ and/or use gtk-doc annotations.  -->
   <package name="gobject-2.0"/>
   <namespace name="GtkFrob"
              version="1.0"
-             shared-library="libgtkfrob.so"
+             shared-library="cyggtkfrob.dll"
              c:prefix="Gtk">
     <function name="language_manager_get_default"
               c:identifier="gtk_frob_language_manager_get_default">
--- origsrc/gobject-introspection-0.6.10/tests/scanner/GtkFrob-1.0-expected.tgir	2010-04-18 13:13:02.000000000 -0500
+++ src/gobject-introspection-0.6.10/tests/scanner/GtkFrob-1.0-expected.tgir	2010-05-13 00:31:08.505326000 -0500
@@ -5,7 +5,7 @@
             xmlns:glib="http://www.gtk.org/introspection/glib/1.0">
   <include name="GObject" version="2.0"/>
   <include name="GLib" version="2.0"/>
-  <namespace name="GtkFrob" version="1.0" shared-library="libgtkfrob.so" c:prefix="Gtk">
+  <namespace name="GtkFrob" version="1.0" shared-library="cyggtkfrob.dll" c:prefix="Gtk">
     <function name="language_manager_get_default" c:identifier="gtk_frob_language_manager_get_default">
       <return-value transfer-ownership="none">
         <type name="none"/>
--- origsrc/gobject-introspection-0.6.10/tests/scanner/utility-1.0-expected.gir	2010-01-21 12:36:18.000000000 -0600
+++ src/gobject-introspection-0.6.10/tests/scanner/utility-1.0-expected.gir	2010-05-13 00:30:12.936147600 -0500
@@ -11,7 +11,7 @@ and/or use gtk-doc annotations.  -->
   <package name="gobject-2.0"/>
   <namespace name="utility"
              version="1.0"
-             shared-library="libutility.so"
+             shared-library="cygutility.dll"
              c:prefix="utility">
     <alias name="Glyph" target="uint32" c:type="UtilityGlyph"/>
     <record name="Buffer" c:type="UtilityBuffer">
--- origsrc/gobject-introspection-0.6.10/tests/scanner/utility-1.0-expected.tgir	2010-04-18 13:13:01.000000000 -0500
+++ src/gobject-introspection-0.6.10/tests/scanner/utility-1.0-expected.tgir	2010-05-13 00:31:15.826744800 -0500
@@ -5,7 +5,7 @@
             xmlns:glib="http://www.gtk.org/introspection/glib/1.0">
   <include name="GObject" version="2.0"/>
   <include name="GLib" version="2.0"/>
-  <namespace name="utility" version="1.0" shared-library="libutility.so" c:prefix="utility">
+  <namespace name="utility" version="1.0" shared-library="cygutility.dll" c:prefix="utility">
     <record name="Buffer">
       <field name="data" writable="1">
         <type name="any"/>
