--- origsrc/gobject-introspection-0.6.7/configure.ac	2009-12-10 13:21:02.000000000 -0600
+++ src/gobject-introspection-0.6.7/configure.ac	2010-01-21 03:44:53.766842400 -0600
@@ -23,14 +23,22 @@ AC_CONFIG_MACRO_DIR([m4])
 AC_CANONICAL_HOST
 AC_MSG_CHECKING([for Win32])
 case "$host" in
+*-*-cygwin*)
+	platform_win32=yes
+	os_win32=no
+	;;
 *-*-mingw*)
+	platform_win32=yes
 	os_win32=yes
 	;;
 *)
+	platform_win32=no
 	os_win32=no
 	;;
 esac
+
 AC_MSG_RESULT([$os_win32])
+AM_CONDITIONAL(PLATFORM_WIN32, [test "$platform_win32" = "yes"])
 AM_CONDITIONAL(OS_WIN32, [test "$os_win32" = "yes"])
 
 # Checks for programs.
--- origsrc/gobject-introspection-0.6.7/gir/Makefile.am	2009-12-14 17:00:16.000000000 -0600
+++ src/gobject-introspection-0.6.7/gir/Makefile.am	2010-01-21 03:44:53.773842800 -0600
@@ -161,7 +161,7 @@ libgirepository_everything_1_0_la_CFLAGS
 libgirepository_everything_1_0_la_LDFLAGS = -version-info $(LT_VERSION)
 libgirepository_everything_1_0_la_LIBADD = $(GOBJECT_LIBS)
 
-if OS_WIN32
+if PLATFORM_WIN32
 libgirepository_everything_1_0_la_LDFLAGS += -no-undefined
 endif
 
--- origsrc/gobject-introspection-0.6.7/giscanner/Makefile.am	2009-09-19 12:27:48.000000000 -0500
+++ src/gobject-introspection-0.6.7/giscanner/Makefile.am	2010-01-21 03:44:53.778843100 -0600
@@ -67,6 +67,10 @@ if OS_WIN32
 # Yuck. Probably there is a way to get this from Python, but I don't
 # know how. Use -Wl to avoid libtool crack.
 _giscanner_la_LDFLAGS += -Wl,$(pyexecdir)/../../libs/libpython25.a -no-undefined
+else
+if PLATFORM_WIN32
+_giscanner_la_LDFLAGS += -L$(libdir)/python$(PYTHON_VERSION)/config -lpython$(PYTHON_VERSION) -no-undefined
+endif
 endif
 
 _giscanner_la_SOURCES = giscannermodule.c
--- origsrc/gobject-introspection-0.6.7/tests/invoke/Makefile.am	2009-09-19 12:27:48.000000000 -0500
+++ src/gobject-introspection-0.6.7/tests/invoke/Makefile.am	2010-01-21 03:44:53.783843400 -0600
@@ -12,7 +12,7 @@ testfns_la_CFLAGS = $(GIREPO_CFLAGS) -I$
 testfns_la_LDFLAGS = -module -avoid-version
 testfns_la_LIBADD = $(top_builddir)/girepository/libgirepository-1.0.la $(GIREPO_LIBS)
 
-if OS_WIN32
+if PLATFORM_WIN32
 testfns_la_LDFLAGS += -no-undefined
 endif
 
--- origsrc/gobject-introspection-0.6.7/tests/scanner/Makefile.am	2009-12-14 17:00:16.000000000 -0600
+++ src/gobject-introspection-0.6.7/tests/scanner/Makefile.am	2010-01-21 03:44:53.790843800 -0600
@@ -23,8 +23,9 @@ libfoo_la_SOURCES = $(srcdir)/foo.c $(sr
 libutility_la_SOURCES = $(srcdir)/utility.c $(srcdir)/utility.h
 libgtkfrob_la_SOURCES = $(srcdir)/gtkfrob.c $(srcdir)/gtkfrob.h
 
-if OS_WIN32
+if PLATFORM_WIN32
 AM_LDFLAGS += -no-undefined
+libfoo_la_LIBADD = $(top_builddir)/girepository/libgirepository-1.0.la
 endif
 
 # .gir --[scanner]-> .typelib --[generate]-> .tgir
@@ -84,10 +85,10 @@
 barapp_LDADD = $(top_builddir)/girepository/libgirepository-1.0.la
 barapp_LDFLAGS = -export-dynamic
 
-BarApp-1.0.gir: barapp
+BarApp-1.0.gir: barapp$(EXEEXT)
 BarApp_1_0_gir_PACKAGES = gobject-2.0
-BarApp_1_0_gir_PROGRAM = ./barapp
+BarApp_1_0_gir_PROGRAM = ./barapp$(EXEEXT)
 BarApp_1_0_gir_INCLUDES = GObject-2.0
 BarApp_1_0_gir_FILES = $(barapp_SOURCES)
 BarApp_1_0_gir_SCANNERFLAGS = --strip-prefix=Bar
