--- origsrc/gobject-introspection-0.9.5/giscanner/Makefile.am	2010-09-10 08:20:57.000000000 -0500
+++ src/gobject-introspection-0.9.5/giscanner/Makefile.am	2010-09-13 05:21:24.395091600 -0500
@@ -68,6 +68,8 @@ if OS_WIN32
 # Yuck. Probably there is a way to get this from Python, but I don't
 # know how. Use -Wl to avoid libtool crack.
 _giscanner_la_LDFLAGS += -Wl,$(pyexecdir)/../../libs/libpython25.a -no-undefined
+else
+_giscanner_la_LDFLAGS += -L$(libdir)/python$(PYTHON_VERSION)/config -lpython$(PYTHON_VERSION) -no-undefined
 endif
 
 _giscanner_la_SOURCES = giscannermodule.c
--- origsrc/gobject-introspection-0.9.5/tests/Makefile.am	2010-09-10 08:20:57.000000000 -0500
+++ src/gobject-introspection-0.9.5/tests/Makefile.am	2010-09-13 06:26:55.984247700 -0500
@@ -7,14 +7,18 @@ BUILT_SOURCES=
 CLEANFILES=
 
 INCLUDES = $(GOBJECT_CFLAGS)
-LIBADD = $(GOBJECT_LIBS)
 
 testlib_LTLIBRARIES = libeverything-1.0.la libgimarshallingtests-1.0.la
 testlibdir=$(prefix)/unused
 install-testlibLTLIBRARIES: # prevent it from being installed
 
 libeverything_1_0_la_SOURCES = everything.c
+libeverything_1_0_la_LDFLAGS = -no-undefined
+libeverything_1_0_la_LIBADD = $(GOBJECT_LIBS)
+
 libgimarshallingtests_1_0_la_SOURCES = gimarshallingtests.c
+libgimarshallingtests_1_0_la_LDFLAGS = -no-undefined
+libgimarshallingtests_1_0_la_LIBADD = $(GOBJECT_LIBS)
 
 EXTRA_DIST += gimarshallingtests.h
 
--- origsrc/gobject-introspection-0.9.5/tests/offsets/Makefile.am	2010-09-10 08:20:57.000000000 -0500
+++ src/gobject-introspection-0.9.5/tests/offsets/Makefile.am	2010-09-13 05:21:32.141534600 -0500
@@ -18,7 +18,7 @@ liboffsets_la_SOURCES = \
 	offsets.c
 liboffsets_la_CPPFLAGS = $(GIREPO_CFLAGS)
 # dummy rpath to get built dynamically (huh?)
-liboffsets_la_LDFLAGS = -avoid-version -rpath $(libdir)
+liboffsets_la_LDFLAGS = -avoid-version -rpath $(libdir) -no-undefined
 
 Offsets-1.0.gir: liboffsets.la offsets.h
 Offsets_1_0_gir_INCLUDES = GObject-2.0
--- origsrc/gobject-introspection-0.9.5/tests/scanner/Makefile.am	2010-09-10 09:49:36.000000000 -0500
+++ src/gobject-introspection-0.9.5/tests/scanner/Makefile.am	2010-09-13 05:22:42.819973800 -0500
@@ -17,7 +17,7 @@ install-testlibLTLIBRARIES: # prevent it
 
 AM_CPPFLAGS = -I$(top_srcdir)/girepository
 AM_CFLAGS = $(GIO_CFLAGS) $(GOBJECT_CFLAGS) $(GTHREAD_CFLAGS)
-AM_LDFLAGS = -avoid-version
+AM_LDFLAGS = -avoid-version -no-undefined
 LIBS = $(GOBJECT_LIBS) $(GTHREAD_LIBS)
 
 libannotation_la_SOURCES = $(srcdir)/annotation.c $(srcdir)/annotation.h
@@ -37,10 +37,6 @@ libregress_la_CFLAGS = $(AM_CFLAGS) $(CA
 libregress_la_LDFLAGS = $(AM_LDFLAGS) $(CAIRO_LIBS) 
 
 
-if OS_WIN32
-AM_LDFLAGS += -no-undefined
-endif
-
 # .gir --[scanner]-> .typelib
 GIRS =
 TYPELIBS = $(GIRS:.gir=.typelib)
@@ -101,9 +97,9 @@ barapp_SOURCES = $(srcdir)/barapp.c $(sr
 barapp_LDADD = $(top_builddir)/girepository/libgirepository-1.0.la
 barapp_LDFLAGS = -export-dynamic
 
-Bar-1.0.gir: barapp
+Bar-1.0.gir: barapp$(EXEEXT)
 Bar_1_0_gir_PACKAGES = gobject-2.0
-Bar_1_0_gir_PROGRAM = ./barapp
+Bar_1_0_gir_PROGRAM = ./barapp$(EXEEXT)
 Bar_1_0_gir_INCLUDES = GObject-2.0
 Bar_1_0_gir_FILES = $(barapp_SOURCES)
 Bar_1_0_gir_SCANNERFLAGS = --accept-unprefixed
