--- origsrc/gobject-introspection-0.9.5/common.mk	2010-06-01 14:48:36.000000000 -0500
+++ src/gobject-introspection-0.9.5/common.mk	2010-09-13 05:15:45.530132100 -0500
@@ -11,7 +11,7 @@ INTROSPECTION_SCANNER = \
         PYTHONPATH=$(top_builddir):$(top_srcdir) \
 	UNINSTALLED_INTROSPECTION_SRCDIR=$(top_srcdir) \
 	UNINSTALLED_INTROSPECTION_BUILDDIR=$(top_builddir) \
-    $(top_builddir)/tools/g-ir-scanner$(EXEEXT)
+    $(top_builddir)/tools/g-ir-scanner
 INTROSPECTION_SCANNER_ARGS = \
     --verbose \
     -I$(top_srcdir) \
--- origsrc/gobject-introspection-0.9.5/gobject-introspection-1.0.pc.in	2010-09-10 08:20:57.000000000 -0500
+++ src/gobject-introspection-0.9.5/gobject-introspection-1.0.pc.in	2010-09-13 05:16:06.446125800 -0500
@@ -7,8 +7,8 @@ datadir=@datadir@
 includedir=@includedir@
 
 g_ir_scanner=${bindir}/g-ir-scanner
-g_ir_compiler=${bindir}/g-ir-compiler
-g_ir_generate=${bindir}/g-ir-generate
+g_ir_compiler=${bindir}/g-ir-compiler@EXEEXT@
+g_ir_generate=${bindir}/g-ir-generate@EXEEXT@
 gidatadir=${datadir}/gobject-introspection-1.0
 girdir=${datadir}/gir-1.0
 typelibdir=${libdir}/girepository-1.0
--- origsrc/gobject-introspection-0.9.5/gobject-introspection-no-export-1.0.pc.in	2009-12-08 07:57:17.000000000 -0600
+++ src/gobject-introspection-0.9.5/gobject-introspection-no-export-1.0.pc.in	2010-09-13 05:15:45.545133000 -0500
@@ -7,8 +7,8 @@ datadir=@datadir@
 includedir=@includedir@
 
 g_ir_scanner=${bindir}/g-ir-scanner
-g_ir_compiler=${bindir}/g-ir-compiler
-g_ir_generate=${bindir}/g-ir-generate
+g_ir_compiler=${bindir}/g-ir-compiler@EXEEXT@
+g_ir_generate=${bindir}/g-ir-generate@EXEEXT@
 girdir=${datadir}/gir-1.0
 typelibdir=${libdir}/girepository-1.0
 
