--- origsrc/gobject-introspection-0.9.5/tests/Makefile.am	2010-09-10 08:20:57.000000000 -0500
+++ src/gobject-introspection-0.9.5/tests/Makefile.am	2010-09-13 06:26:55.984247700 -0500
@@ -22,7 +26,7 @@ BUILT_SOURCES += everything.c everything
 
 CLEANFILES += $(BUILT_SOURCES) everything-stamp.h
 everything-stamp.h: Makefile
-	$(AM_V_GEN) $(top_builddir)/tools/g-ir-scanner --generate-typelib-tests=Everything,everything.h,everything.c
+	$(AM_V_GEN) $(INTROSPECTION_SCANNER) --generate-typelib-tests=Everything,everything.h,everything.c
 	touch $@
 
 everything.c: everything-stamp.h
@@ -43,7 +47,7 @@ GIMarshallingTests-1.0.gir: libgimarshal
 	--namespace=GIMarshallingTests --symbol-prefix=gi_marshalling_tests --nsversion=1.0 \
 	--add-include-path=$(top_builddir)/gir --include=Gio-2.0 \
 	--library=libgimarshallingtests-1.0.la  --output=$@ \
-	gimarshallingtests.h gimarshallingtests.c
+	$(srcdir)/gimarshallingtests.h $(srcdir)/gimarshallingtests.c
 
 %.typelib: %.gir
 	$(AM_V_GEN) $(INTROSPECTION_COMPILER) $(INTROSPECTION_COMPILER_ARGS) $< -o $@
@@ -52,5 +56,9 @@ check: Everything-1.0.typelib GIMarshall
 
 install-data-local:
 	for src in everything.c everything.h gimarshallingtests.c gimarshallingtests.h; do \
+      if test -f $$src; then \
 	  install -m 644 -D $$src $(DESTDIR)$(datadir)/gobject-introspection-1.0/tests/$$src; \
+      else \
+	  install -m 644 -D $(srcdir)/$$src $(DESTDIR)$(datadir)/gobject-introspection-1.0/tests/$$src; \
+      fi; \
 	done
--- origsrc/gobject-introspection-0.9.5/tests/scanner/Makefile.am	2010-09-13 05:23:44.835318200 -0500
+++ src/gobject-introspection-0.9.5/tests/scanner/Makefile.am	2010-09-13 06:30:47.457953700 -0500
@@ -29,8 +29,8 @@ libgtkfrob_la_SOURCES = $(srcdir)/gtkfro
 libregress_la_SOURCES = $(srcdir)/regress.c $(srcdir)/regress.h
 
 install-data-local:
-	for src in $(libregress_la_SOURCES); do \
-	  install -m 644 -D $$src $(DESTDIR)$(datadir)/gobject-introspection-1.0/tests/$$src; \
+	for src in regress.c regress.h; do \
+	  install -m 644 -D $(srcdir)/$$src $(DESTDIR)$(datadir)/gobject-introspection-1.0/tests/$$src; \
 	done
 
 libregress_la_CFLAGS = $(AM_CFLAGS) $(CAIRO_CFLAGS)
