--- origsrc/gobject-introspection-0.9.5/tests/scanner/Annotation-1.0-expected.gir	2010-09-10 08:20:57.000000000 -0500
+++ src/gobject-introspection-0.9.5/tests/scanner/Annotation-1.0-expected.gir	2010-09-13 05:13:28.506902600 -0500
@@ -12,7 +12,7 @@ and/or use gtk-doc annotations.  -->
   <package name="gobject-2.0"/>
   <namespace name="Annotation"
              version="1.0"
-             shared-library="libannotation.so"
+             shared-library="cygannotation.dll"
              c:identifier-prefixes="Annotation"
              c:symbol-prefixes="annotation">
     <callback name="Callback" c:type="AnnotationCallback">
--- origsrc/gobject-introspection-0.9.5/tests/scanner/Foo-1.0-expected.gir	2010-09-10 08:20:57.000000000 -0500
+++ src/gobject-introspection-0.9.5/tests/scanner/Foo-1.0-expected.gir	2010-09-13 05:13:52.545074900 -0500
@@ -14,7 +14,7 @@ and/or use gtk-doc annotations.  -->
   <c:include name="foo.h"/>
   <namespace name="Foo"
              version="1.0"
-             shared-library="libfoo.so"
+             shared-library="cygfoo.dll"
              c:identifier-prefixes="Foo"
              c:symbol-prefixes="foo">
     <alias name="ObjectCookie" c:type="FooObjectCookie">
--- origsrc/gobject-introspection-0.9.5/tests/scanner/GtkFrob-1.0-expected.gir	2010-09-10 08:20:57.000000000 -0500
+++ src/gobject-introspection-0.9.5/tests/scanner/GtkFrob-1.0-expected.gir	2010-09-13 05:13:58.769430900 -0500
@@ -11,7 +11,7 @@ and/or use gtk-doc annotations.  -->
   <package name="gobject-2.0"/>
   <namespace name="GtkFrob"
              version="1.0"
-             shared-library="libgtkfrob.so"
+             shared-library="cyggtkfrob.dll"
              c:identifier-prefixes="Gtk"
              c:symbol-prefixes="gtk_frob">
     <function name="language_manager_get_default"
--- origsrc/gobject-introspection-0.9.5/tests/scanner/Regress-1.0-expected.gir	2010-09-10 08:20:57.000000000 -0500
+++ src/gobject-introspection-0.9.5/tests/scanner/Regress-1.0-expected.gir	2010-09-13 05:14:04.538760900 -0500
@@ -12,7 +12,7 @@ and/or use gtk-doc annotations.  -->
   <include name="cairo" version="1.0"/>
   <namespace name="Regress"
              version="1.0"
-             shared-library="libregress.so"
+             shared-library="cygregress.dll"
              c:identifier-prefixes="Regress"
              c:symbol-prefixes="regress">
     <constant name="DOUBLE_CONSTANT" value="44.220000">
--- origsrc/gobject-introspection-0.9.5/tests/scanner/TestInherit-1.0-expected.gir	2010-09-10 08:20:57.000000000 -0500
+++ src/gobject-introspection-0.9.5/tests/scanner/TestInherit-1.0-expected.gir	2010-09-13 05:14:11.219143000 -0500
@@ -12,7 +12,7 @@ and/or use gtk-doc annotations.  -->
   <package name="gobject-2.0"/>
   <namespace name="TestInherit"
              version="1.0"
-             shared-library="libtestinherit.so"
+             shared-library="cygtestinherit.dll"
              c:identifier-prefixes="TestInherit"
              c:symbol-prefixes="test_inherit">
     <class name="Drawable"
--- origsrc/gobject-introspection-0.9.5/tests/scanner/Utility-1.0-expected.gir	2010-09-10 08:20:57.000000000 -0500
+++ src/gobject-introspection-0.9.5/tests/scanner/Utility-1.0-expected.gir	2010-09-13 05:14:16.893467600 -0500
@@ -11,7 +11,7 @@ and/or use gtk-doc annotations.  -->
   <package name="gobject-2.0"/>
   <namespace name="Utility"
              version="1.0"
-             shared-library="libutility.so"
+             shared-library="cygutility.dll"
              c:identifier-prefixes="Utility"
              c:symbol-prefixes="utility">
     <alias name="Glyph" c:type="UtilityGlyph">
