--- origsrc/gobject-introspection-1.30.0/Makefile-gir.am	2011-09-03 13:36:47.000000000 -0500
+++ src/gobject-introspection-1.30.0/Makefile-gir.am	2011-11-15 19:48:04.068221400 -0600
@@ -26,17 +26,17 @@ CLEANFILES += gir/cairo-1.0.gir
 EXTRA_DIST += gir/cairo-1.0.gir.in
 
 # Doesn't seem to work to do gir/%.typelib =(
-gir/cairo-1.0.typelib: g-ir-compiler
-gir/DBus-1.0.typelib: g-ir-compiler
-gir/DBusGLib-1.0.typelib: g-ir-compiler
-gir/fontconfig-2.0.typelib: g-ir-compiler
-gir/freetype2-2.0.typelib: g-ir-compiler
-gir/GL-1.0.typelib: g-ir-compiler
-gir/libxml2-2.0.typelib: g-ir-compiler
-gir/xft-2.0.typelib: g-ir-compiler
-gir/xlib-2.0.typelib: g-ir-compiler
-gir/xfixes-4.0.typelib: g-ir-compiler
-gir/xrandr-1.3.typelib: g-ir-compiler
+gir/cairo-1.0.typelib: g-ir-compiler$(EXEEXT)
+gir/DBus-1.0.typelib: g-ir-compiler$(EXEEXT)
+gir/DBusGLib-1.0.typelib: g-ir-compiler$(EXEEXT)
+gir/fontconfig-2.0.typelib: g-ir-compiler$(EXEEXT)
+gir/freetype2-2.0.typelib: g-ir-compiler$(EXEEXT)
+gir/GL-1.0.typelib: g-ir-compiler$(EXEEXT)
+gir/libxml2-2.0.typelib: g-ir-compiler$(EXEEXT)
+gir/xft-2.0.typelib: g-ir-compiler$(EXEEXT)
+gir/xlib-2.0.typelib: g-ir-compiler$(EXEEXT)
+gir/xfixes-4.0.typelib: g-ir-compiler$(EXEEXT)
+gir/xrandr-1.3.typelib: g-ir-compiler$(EXEEXT)
 
 # glib with source (to enable documentation)
 if WITH_GLIBSRC
--- origsrc/gobject-introspection-1.30.0/gobject-introspection-1.0.pc.in	2011-06-10 10:36:50.000000000 -0500
+++ src/gobject-introspection-1.30.0/gobject-introspection-1.0.pc.in	2011-11-15 19:48:33.608262200 -0600
@@ -7,8 +7,8 @@ datadir=@datadir@
 includedir=@includedir@
 
 g_ir_scanner=${bindir}/g-ir-scanner
-g_ir_compiler=${bindir}/g-ir-compiler
-g_ir_generate=${bindir}/g-ir-generate
+g_ir_compiler=${bindir}/g-ir-compiler@EXEEXT@
+g_ir_generate=${bindir}/g-ir-generate@EXEEXT@
 gidatadir=${datadir}/gobject-introspection-1.0
 girdir=${datadir}/gir-1.0
 typelibdir=${libdir}/girepository-1.0
--- origsrc/gobject-introspection-1.30.0/gobject-introspection-no-export-1.0.pc.in	2011-06-10 10:36:50.000000000 -0500
+++ src/gobject-introspection-1.30.0/gobject-introspection-no-export-1.0.pc.in	2011-11-15 19:48:53.148289200 -0600
@@ -7,8 +7,8 @@ datadir=@datadir@
 includedir=@includedir@
 
 g_ir_scanner=${bindir}/g-ir-scanner
-g_ir_compiler=${bindir}/g-ir-compiler
-g_ir_generate=${bindir}/g-ir-generate
+g_ir_compiler=${bindir}/g-ir-compiler@EXEEXT@
+g_ir_generate=${bindir}/g-ir-generate@EXEEXT@
 girdir=${datadir}/gir-1.0
 typelibdir=${libdir}/girepository-1.0
 
--- origsrc/gobject-introspection-1.30.0/tests/scanner/Makefile.am	2011-09-09 15:48:44.000000000 -0500
+++ src/gobject-introspection-1.30.0/tests/scanner/Makefile.am	2011-11-15 19:40:38.187606100 -0600
@@ -106,9 +106,9 @@ barapp_SOURCES = $(srcdir)/barapp.c $(sr
 barapp_LDADD = $(top_builddir)/libgirepository-1.0.la
 barapp_LDFLAGS = -export-dynamic
 
-Bar-1.0.gir: barapp
+Bar-1.0.gir: barapp$(EXEEXT)
 Bar_1_0_gir_PACKAGES = gobject-2.0
-Bar_1_0_gir_PROGRAM = ./barapp
+Bar_1_0_gir_PROGRAM = ./barapp$(EXEEXT)
 Bar_1_0_gir_INCLUDES = GObject-2.0
 Bar_1_0_gir_FILES = $(barapp_SOURCES)
 Bar_1_0_gir_SCANNERFLAGS = --accept-unprefixed
